releasing_guide.md

```diff
@@ -18,9 +18,13 @@ To publish a new LumiSpy release do the following steps:
 - Create a tag e.g. `git tag -a v0.1.1 -m "LumiSpy version 0.1.1"`
 - Push tag to user fork for a test run `git push origin v0.1.1`. Will run the release
   workflow without uploading to PyPi
-- Push tag to LumiSpy repository to trigger release `git push upstream v0.1.1`
-  (this triggers the GitHub action to create the sdist and wheel and upload to
-  PyPi automatically). :warning: this is a point of no return :warning:
+- Push tag to LumiSpy repository to trigger release `git push upstream v0.1.1`. 
+  :warning: this is a point of no return :warning:
+  - triggers the GitHub action to create the sdist and wheel and upload them to
```

Comment:
Maybe it would be better to create the github release after the packages are upload to pypi, in case the release workflow fails at some point, as this would to push again the tag without to worry about the Zenodo entry being created on the GitHub release creation.
I added some additional notes in https://github.com/hyperspy/rosettasciio/pull/126

Reply:
Valid point, maybe we should indeed adapt our release workflow to match the one of rosettasciio.